pkg/net/libp2p/key.go

```diff
@@ -27,15 +27,32 @@ func operatorPrivateKeyToNetworkKeyPair(operatorPrivateKey *operator.PrivateKey)
 		return nil, nil, fmt.Errorf("libp2p supports only secp256k1 operator keys")
 	}
 
-	// Libp2p keys are actually btcec keys under the hood.
-	btcecPrivateKey, btcecPublicKey := btcec.PrivKeyFromBytes(
-		DefaultCurve, operatorPrivateKey.D.Bytes(),
+	// Note that `libp2pcrypto.UnmarshalSecp256k1PrivateKey` uses the secp256k1
+	// implementation provided by decred underneath
+	privKey, err := libp2pcrypto.UnmarshalSecp256k1PrivateKey(
+		operatorPrivateKey.D.Bytes(),
 	)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pubKey := privKey.GetPublic()
+
+	privateNetworkKey, ok := privKey.(*libp2pcrypto.Secp256k1PrivateKey)
+	if !ok {
+		return nil, nil, fmt.Errorf(
+			"network private key is not an instance of the libp2p secp256k1 private key",
+		)
+	}
 
-	networkPrivateKey := libp2pcrypto.Secp256k1PrivateKey(*btcecPrivateKey)
-	networkPublicKey := libp2pcrypto.Secp256k1PublicKey(*btcecPublicKey)
+	publicNetworkKey, ok := pubKey.(*libp2pcrypto.Secp256k1PublicKey)
```

Comment:
s/publicNetworkKey/networkPublicKey

Reply:
Done in https://github.com/keep-network/keep-core/commit/86d4e21916f151b85db250a1431977fe4ca6ab8a.